src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/hdc/HdcApiClient.kt

```diff
@@ -1,21 +1,41 @@
 package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service.hdc
 
+import jakarta.persistence.EntityNotFoundException
+import org.slf4j.LoggerFactory
 import org.springframework.beans.factory.annotation.Qualifier
+import org.springframework.http.HttpStatus.NOT_FOUND
 import org.springframework.http.MediaType
 import org.springframework.stereotype.Service
 import org.springframework.web.reactive.function.client.WebClient
-import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.typeReference
+import org.springframework.web.reactive.function.client.WebClientResponseException
+import reactor.core.publisher.Mono
 
 @Service
 class HdcApiClient(@Qualifier("oauthHdcApiClient") val hdcApiWebClient: WebClient) {
 
+  companion object {
+    private val log = LoggerFactory.getLogger(this::class.java)
+  }
+
   fun getByBookingId(bookingId: Long): HdcLicenceData? {
     return hdcApiWebClient
       .get()
       .uri("/licence/hdc/$bookingId")
       .accept(MediaType.APPLICATION_JSON)
       .retrieve()
-      .bodyToMono(typeReference<HdcLicenceData>())
+      .bodyToMono(HdcLicenceData::class.java)
+      .onErrorResume { propagate404Response(it, bookingId) }
       .block()
   }
+
+  private fun <API_RESPONSE_BODY_TYPE> propagate404Response(exception: Throwable, bookingId: Long): Mono<API_RESPONSE_BODY_TYPE> =
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of. -->
**note:** ‏I'm hoping this is what you meant by propagating - if there is an exception and it's a NOT_FOUND, this will throw a 404 and not 500 - integration test included.